segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/PriorityCalculator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.logs;
+
+import io.pravega.segmentstore.contracts.SegmentType;
+import io.pravega.segmentstore.server.logs.operations.OperationPriority;
+import io.pravega.segmentstore.server.logs.operations.OperationType;
+
+/**
+ * Calculates {@link OperationPriority} based on various factors.
+ */
+public final class PriorityCalculator {
+
+    /**
+     * Determines an {@link OperationPriority} appropriate for the given {@link SegmentType} and {@link OperationType}.
+     *
+     * <pre>
+     *
+     * SegmentType       | OperationType | Priority
+     * ------------------+---------------+----------------
+     * System & Critical | (any)         | {@link OperationPriority#SystemCritical}
+     * Critical          | (any)         | {@link OperationPriority#Critical}
+     * (any)             | Deletion      | {@link OperationPriority#Critical}
+     * System            | Normal        | {@link OperationPriority#High}
+     * (all other combinations)          | {@link OperationPriority#Normal}
+     *
+     * </pre>
+     *
+     * @param segmentType   {@link SegmentType} that the Operation applies to.
+     * @param operationType Operation's {@link OperationType}.
+     * @return A {@link OperationPriority}.
+     */
+    public OperationPriority getPriority(SegmentType segmentType, OperationType operationType) {
```

Comment:
True. But I want to leave the door open for future changes in case we want to create different types of priority calculators (using inheritance or implementing interfaces). We may have to refactor this class, but not fix all callsites to switch from static invocations to an instance invocation.

Reply:
We can do that for now.